doc/status-current-code.md

```diff
@@ -0,0 +1,63 @@
+## Current Network Code Overview
+
+The main responsabilities of Network configuration resides between yast2-network 
+and yast2 (NetworkInterfaces) and them are by sure the more complex parts of
+the code.
+
+- NetworkInterfaces (yast2): represent mainly the configured interfaces and is in
+  charge of writing the ifcfg-giles. The numbers of bugs in this area are not
+  so big, we have been dealing mostly with some normalization of the config or
+  better defaults like for wireless configuration.
+
+- LanItems (yast2-network): It was introduced in 2007 [1] for unifying all the
+  maps in a single list.
+
+  So basically each LanItems Item has the information about its (hwinfo, ifcfg
+  file, [udev rules](udev-rules-implementation.org), and table description 
+  (the overview))
+
+  The match between these different objects / hashes is based on the interface 
+  or device name, and as we can see it has been one of me most common point of 
+  failures in network configuration.
+
+  Of course, naming devices is a known problems in Networking an that was the
+  reason of the new predictable network interfaces naming. Something we are
+  not prepared yet but which is already used in TW (we still offer to rename
+  the interfaces based on [udev rules](udev_rules.md) even when does not work
+  on this scenario)
```

Comment:
well not prepared ... there is only one problematic part ... and that is transition (or creating) of configuration from instsys to installed target.